gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +734,689 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+/*                              SetScale()                              */
+/************************************************************************/
+
+CPLErr FITSRasterBand::SetScale( double dfNewValue )
+
+{
+    poDS->LoadGeoreferencingAndPamIfNeeded();
+
+    if( !bHaveOffsetScale || dfNewValue != dfScale )
+        poDS->bMetadataChanged = true;
+
+    bHaveOffsetScale = true;
+    dfScale = dfNewValue;
+    return CE_None;
+}
+
+/************************************************************************/
+/*                           GetNoDataValue()                           */
+/************************************************************************/
+
+double FITSRasterBand::GetNoDataValue( int * pbSuccess )
+
+{
+    poDS->LoadGeoreferencingAndPamIfNeeded();
+
+    if( bNoDataSet )
+    {
+        if( pbSuccess )
+            *pbSuccess = TRUE;
+
+        return dfNoDataValue;
+    }
+
+    if( poDS->bNoDataSet )
+    {
+        if( pbSuccess )
+            *pbSuccess = TRUE;
+
+        return poDS->dfNoDataValue;
+    }
+
+    return GDALPamRasterBand::GetNoDataValue( pbSuccess );
+}
+
+/************************************************************************/
+/*                           SetNoDataValue()                           */
+/************************************************************************/
+
+CPLErr FITSRasterBand::SetNoDataValue( double dfNoData )
+
+{
+    poDS->LoadGeoreferencingAndPamIfNeeded();
+
+    if( poDS->bNoDataSet && poDS->dfNoDataValue == dfNoData )
+    {
+        bNoDataSet = true;
+        dfNoDataValue = dfNoData;
+        return CE_None;
+    }
+
+    poDS->bNoDataSet = true;
+    poDS->dfNoDataValue = dfNoData;
+
+    poDS->bNoDataChanged = true;
+
+    bNoDataSet = true;
+    dfNoDataValue = dfNoData;
+    return CE_None;
+}
+
+/************************************************************************/
+/*                        DeleteNoDataValue()                           */
+/************************************************************************/
+
+CPLErr FITSRasterBand::DeleteNoDataValue()
+
+{
+    poDS->LoadGeoreferencingAndPamIfNeeded();
+
+    if( !poDS->bNoDataSet )
+        return CE_None;
+
+    poDS->bNoDataSet = false;
+    poDS->dfNoDataValue = -9999.0;
+
+    poDS->bNoDataChanged = true;
+
+    bNoDataSet = false;
+    dfNoDataValue = -9999.0;
+    return CE_None;
+}
+
+/************************************************************************/
+/*                     LoadGeoreferencingAndPamIfNeeded()               */
+/************************************************************************/
+
+void FITSDataset::LoadGeoreferencingAndPamIfNeeded()
+
+{
+    int status = 0;
+    int bitpix;
+    double dfScale, dfOffset;
+    double crpix1, crpix2, crval1, crval2, cdelt1, cdelt2, pc[4], cd[4];
+    double aRadius, cRadius, invFlattening = 0.0;
+    char target[16], ctype[9];
+    std::string pszGeogName, pszDatumName, projName;
```

Comment:
do not use psz for std::string objects. psz is a *p*ointer to a *s*tring *z*ero terminated, so for C-style strings

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/f17edf20f02f51decbbc735b0dfe0c4219a20111